pandas/tests/io/excel/test_xlsxwriter.py

```diff
@@ -63,3 +67,29 @@ def test_write_append_mode_raises(ext):
     with ensure_clean(ext) as f:
         with pytest.raises(ValueError, match=msg):
             ExcelWriter(f, engine='xlsxwriter', mode='a')
+
+
+def test_merged_cell_custom_objects(ext):
+    # Test that custom object types residing within merged (grouped)
+    # cells are converted to python data types before being passed to
+    # the xlsxwriter package. Test for issue #27006
+
+    # create a custom object type, and place it in a grouped dataframe
+    pixy = PeriodIndex(['2018', '2018', '2018', '2018',
```

Comment:
You can make these even shorter (really just one row of a MI would suffice) to limit the amount of code need to construct. Something like `MultiIndex.from_tuples([(pd.Period('2018'), pd.Period('2018Q1'))])` could replace quite a few lines here

Reply:
I trimmed this down significantly per your suggestion. The only diff vs your example is that MI needs to contain at least 2 Quarters in the same Year so that when put in a DataFrame, the Year index spans multiple columns. This is the 'merged' condition of the cell that was causing the error in the first place. If there's a more efficient way to write it than I have, I'm all ears.